tests/app/views/test_services.py

```diff
@@ -986,6 +986,185 @@ def test_should_not_index_on_service_on_expired_frameworks(self):
             assert_false(search_api_client.index.called)
 
 
+class TestShouldCallSearchApiOnPostStatusUpdate(BaseApplicationTest):
+    def setup(self):
+        super(TestShouldCallSearchApiOnPostStatusUpdate, self).setup()
+        now = datetime.now()
+        self.services = {}
+
+        valid_statuses = [
+            'published',
+            'enabled',
+            'disabled'
+        ]
+
+        with self.app.app_context():
+            db.session.add(
+                Supplier(supplier_id=1, name=u"Supplier 1")
+            )
+
+            for index, status in enumerate(valid_statuses):
+                payload = self.load_example_listing("G6-IaaS")
+
+                # give each service a different id.
+                new_id = int(payload['id']) + index
+                payload['id'] = "{}".format(new_id)
+
+                self.services[status] = payload
+
+                db.session.add(Service(service_id=self.services[status]['id'],
+                                       supplier_id=1,
+                                       updated_at=now,
+                                       status=status,
+                                       created_at=now,
+                                       updated_by="tests",
+                                       framework_id=1,
+                                       updated_reason="test data",
+                                       data=self.services[status]))
+
+            db.session.commit()
+            assert_equal(3, db.session.query(Service).count())
+
+    def _post_update_status(self, service_id, new_status):
+        return self.client.post(
+            '/services/{0}/status/{1}'.format(
+                service_id,
+                new_status
+            ),
+            data=json.dumps(
+                {'update_details': {
+                    'updated_by': 'joeblogs',
+                    'update_reason': 'Change status for unit test'}
+                 }),
+            content_type='application/json'
+        )
+
+    def test_should_index_on_service_status_changed_to_published(self):
+
+        old_status = 'enabled'
+        new_status = 'published'
+
+        search_api_client.index = Mock(return_value=True)
+        search_api_client.delete = Mock(return_value=True)
```

Comment:
I think this would be safer using `mock.patch`. Currently this is leaving the search API client's index and delete methods mocked after the test is done.